### ISSUE TITLE
Improve b2s or replace with custom code (unneded wide strings support)

### ISSUE THREAD

**Issue by mr-tz** [mandiant/flare-floss#867]
binary2strings reads utf8 and wide strings (we don't expect or require the latter for Rust strings extraction).

In `rust-hello64.exe` we see
<img width="366" alt="2023-08-28_15-09-13_ida64" src="https://github.com/mandiant/flare-floss/assets/17606537/268829fe-fd3a-49a0-b607-cd5516859dd2">

Which parsed as wide string starting at offset 0x1400BD04F, b2s sees as:
<img width="173" alt="2023-08-28_15-11-07_pycharm64" src="https://github.com/mandiant/flare-floss/assets/17606537/a4e0aa87-39b6-4b6c-914f-d49daa7bed9d">

We then filter out the wide string and end up with ` args` in the final Rust strings output.

**Comment by mr-tz** [comment]
Yes, currently we only require a function to find UTF-8 strings though.

**Comment by vaijavi** [comment]
Hi maintainers,
I would like to work on this issue. Please let me know if I can take it.
Thanks!

**Comment by mr-tz** [comment]
Go ahead

**Comment by svm1048** [comment]
hello @mr-tz! I do realise that vaijavi wants to work on this bug, i would like to offer my help as well so that it dosent come off as "sniping". Is it okay if i start working on this ?

**Comment by mr-tz** [comment]
I think you can go ahead.

**Comment by svm1048** [comment]
PR #1223
take a look, also I haven't been covered under CLA, I have applied will lyk how that goes!

**Comment by kunalsz** [comment]
@mr-tz not snipping this issue, just some research I did. The regex based UTF8 strings extraction like in #1223 :
- itself encodes UTF8 validity rules 
- allows ASCII printable + \t\r\n unconditionally (as part of the regex)
- finds any match anywher; no explicit start filtering needed
- no explicit maximum match size; in worst-case it can match and allocate/decode a very large string in one go

It is simple and fast but I made a decoder using python’s incremental UTF‑8 decoder as the authority on validity which addresses these shortcomings (and doesn't use hardcoded regex)

Here are the results(after|before):
<img width="857" height="313" alt="Image" src="https://github.com/user-attachments/assets/6c0653aa-e2f2-4f2c-a5c9-bd1ac09f1089" />
There's a slight slowdown cause b2s used c++, while this is a pure python implementation but it does the job pretty well
(before|after)
<img width="936" height="322" alt="Image" src="https://github.com/user-attachments/assets/6811a410-8fbf-415c-b1b1-9b6627d9532a" />

If you like this approach more than the regex one, I can make a PR

**Comment by williballenthin** [comment]
@svm1048 @kunalsz i think it doesn't hurt to have a comparison of the various approaches, to see ensure the behavior is correct, and if so, also performant and maintainable.

i am interested in the following dimensions:
1. do the techniques match exactly the same strings as b2s? what is added/missing?
2. how does runtime compare with b2s? lets use a common .exe from testfiles, if possible.
3. how much code does it take to implement the algorithm (PR diff size)?

would each of you @svm1048 @kunalsz  consider collecting this info and sharing it here? then we can collaborate together.

**Comment by kunalsz** [comment]
@williballenthin My approach gives the following results :

1. The comparison of results before(the current scenario) vs after the implementation of custom utf-8 parsing:

This improved
<img width="1476" height="57" alt="Image" src="https://github.com/user-attachments/assets/ec78071f-795a-42fd-9f45-533f8500554f" />
<img width="1398" height="61" alt="Image" src="https://github.com/user-attachments/assets/42e46ce2-e9f5-4fcf-8091-4fb626bd07b0" />

Many junk strings got removed
<img width="1473" height="118" alt="Image" src="https://github.com/user-attachments/assets/c7505a35-3919-4140-9b87-bcac4d17b32a" />
<img width="1481" height="111" alt="Image" src="https://github.com/user-attachments/assets/4082a927-a121-4016-9e0a-e53b5da69e4e" />
<img width="1557" height="227" alt="Image" src="https://github.com/user-attachments/assets/ca4465c8-f3f2-4cea-befc-010fc5999d03" />

2.  I attached screenshots in my previous comment comparing b2s and my approach, b2s took 1.91s while this one took 3.16s

3. The `utf8strings.py` which I wrote took around 130 lines , while in `extract.py`  8 lines were added and 57 removed

**Comment by kunalsz** [comment]
@williballenthin Should I make a PR for this , if the results look satisfactory ?

**Comment by williballenthin** [comment]
not at this time no. i currently assess this will take more effort than is worth it.
